lib/system/web/tests/keyboard_engine.rs

```diff
@@ -0,0 +1,40 @@
+use wasm_bindgen_test::*;
+use web_sys::{KeyboardEvent, KeyboardEventInit};
+use basegl_system_web::keyboard_engine::KeyboardEngine;
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+#[wasm_bindgen_test]
+fn capture_keypress() {
+    let window = web_sys::window().expect("no global window exists");
+    let document = window.document().expect("should have document on window");
+    let elem = document.create_element("canvas").unwrap();
+
+    let mut ke = KeyboardEngine::new(&elem, false);
+    ke.capture(String::from("a"), Box::new(|| {
+        web_sys::console::log_1(&"Key has been captured".into());
+    }));
+    let mut event_init = KeyboardEventInit::new();
+    event_init.code("a");
+    let event = KeyboardEvent::new_with_keyboard_event_init_dict("keydown", &event_init).unwrap();
+    elem.dispatch_event(&event).unwrap();
+    assert_eq!(2 + 2, 4); // TODO
+}
+
+#[wasm_bindgen_test]
+fn capture_keypress_on_tag_name() {
+    let window = web_sys::window().expect("no global window exists");
+    let document = window.document().expect("should have document on window");
+    let elem = document.create_element("canvas").unwrap();
+    document.first_child().unwrap().append_child(&elem).expect("ok");
+
+    let mut ke = KeyboardEngine::from_tag_name("canvas", false);
+    ke.capture(String::from("b"), Box::new(|| {
+        web_sys::console::log_1(&"Key has been captured".into());
+    }));
+    let mut event_init = KeyboardEventInit::new();
+    event_init.code("b");
+    let event = KeyboardEvent::new_with_keyboard_event_init_dict("keydown", &event_init).unwrap();
+    elem.dispatch_event(&event).unwrap();
+    assert_eq!(2 + 2, 4); // TODO
+}
```

Comment:
Big applause for tests 👏 
Amazing that you are doing them in the right way (emitting events from code and testing results this way)